src/config/default/fetch.sh

```diff
@@ -71,6 +81,41 @@ if [ "${CONTAINER}" == "gate" ]; then
     else
         echo "No SAML IDP pemfile found at ${saml_pem_path}"
     fi
+    if [ -f ${x509_ca_cert_path} ]; then
+        echo "Loading ${x509_ca_cert_path} into ${x509_jks_path}"
+        add_ca_certs ${x509_ca_cert_path} ${x509_jks_path} "ca"
+    else
+        echo "No x509 CA cert found at ${x509_ca_cert_path}"
+    fi
+    if [ -f ${x509_client_cert_path} ]; then
+        echo "Loading ${x509_client_cert_path} into ${x509_jks_path}"
+        add_ca_certs ${x509_client_cert_path} ${x509_jks_path} "client"
+    else
+        echo "No x509 Client cert found at ${x509_client_cert_path}"
+    fi
+
+
+    if [ -f ${x509_nginx_cert_path} ]; then
+        echo "Creating a self-signed CA (EXPIRES IN 360 DAYS) with java keystore: ${x509_jks_path}"
+        echo -e "\n\n\n\n\n\ny\n" | keytool -genkey -keyalg RSA -alias server -keystore keystore.jks -storepass changeit -validity 360 -keysize 2048
+        keytool -importkeystore \
+                -srckeystore keystore.jks \
+                -srcstorepass changeit \
+                -destkeystore "${x509_jks_path}" \
+                -storepass changeit \
+                -srcalias server \
+                -destalias server \
+                -noprompt
+    else
+        echo "No x509 nginx cert found at ${x509_nginx_cert_path}"
+    fi
+fi
+
+if [ "${CONTAINER}" == "nginx" ]; then
+    nginx_conf_path="/opt/spinnaker/config/default/nginx.conf"
+    if [ -f ${nginx_conf_path} ]; then
+        cp ${nginx_conf_path} /etc/nginx/nginx.conf
+    fi
```

Comment:
This comes from the volume mount for custom config on the nginx deployment. Do we still need this?

Reply:
Synced up with Ethan... This is fine!